### ISSUE TITLE
question about paper

### ISSUE THREAD

**Issue by valencebond** [shengliu66/ELR#3]
thanks for your code and work. But I am confused with the statements in paragraph 2 of Section4.1 

> If c [i] is the true class so that y c = 1, the contribution of the ith example to ∇L CE (Θ) c is aligned with −x [i], and gradient descent moves in the direction of x [i].

i quite understand and agree with you in Eq. 2. According to eq 2, the direction of the gradient is based on difference bewteen softmax prediction and groud truth. Both are important for gradient direction, not only denpends on ground truth y c = 1. It is really nice if you can give some explanation on above statement, especially about "is alighed with -x[i]".

thanks!

**Comment by valencebond** [comment]
i also compare the performance between ELR and cross-entropy baseline where lamda of train_loss is set to 0. But there is no much performance difference. ELR log is 
[info.log](https://github.com/shengliu66/ELR/files/5412709/info.log) and baseline log is 
[info_baseline.log](https://github.com/shengliu66/ELR/files/5412711/info_baseline.log). And the best performance is 0.85 for baseline and 0.86 for ELR.

**Comment by shengliu66** [comment]
Can you tell me what is the config file you are using? what is the noise level?

**Comment by valencebond** [comment]
hi @shengliu66, the noise level is 0.4 and config file is config_cifar10.json. i only change the lambda to reimplement the baseline.

**Comment by shengliu66** [comment]
I checked your log file, looks like you set lambda to 1, can you tell me the reason why you are doing it?

**Comment by valencebond** [comment]
i also visualize the dot product between pi and ti in eq (6) for clear label and wrong label separately in CIFAR10 with noisy level 0.4. 
<img width="370" alt="Screen Shot 2020-10-21 at 9 54 31 AM" src="https://user-images.githubusercontent.com/44313446/96663863-2ad2c900-1384-11eb-9a74-57f3d8079e60.png">

according to the plot, the intuitive explanation of your methods is to add a different negative loss for clear samples and noisy samples. This negative loss is more serious for clear label after early-training stage which is not consistent with the paper. Please correct me if there is some wrong.  Or can you give a intuitive explanation for your ELR loss.

**Comment by valencebond** [comment]
> I checked your log file, looks like you set lambda to 1, can you tell me the reason for doing it? The default value in the config file was 3

i check the issue 2 and according to your advice.

**Comment by shengliu66** [comment]
it is for asymmetric noise if you read it more clearly. I don't understand the plot what does red and the blue line represents for? and we said about the gradient, not the loss.

**Comment by valencebond** [comment]
sorry for the confusion, the blue line is for clear label and red line is for noisy label.  The value is inner product between pi and ti in eq (6). Considering the log(1 - dot product)  < 0 , the bigger the value, the smaller ELR loss. 

<img width="582" alt="Screen Shot 2020-10-21 at 10 13 59 AM" src="https://user-images.githubusercontent.com/44313446/96665293-2956d000-1387-11eb-9046-75f33e9d87e1.png">

Your motivation is based on gradient, i totally agree with it. But in my opinion, if the gradient can reflect something meaningful, then the loss can also be. Paper said 

> Thus, ELR fulﬁls the two desired properties outlined at the end of Section 3: boosting the gradient of examples with clean labels, and neutralizing the gradient of the examples with false labels.

But according to the plot, ELR loss cancel out clear label loss more than noisy label, which make me confusing.

**Comment by valencebond** [comment]
Or the problem is that I shouldn't consider this issue from the perspective of loss at all. But it seems make no sense.

**Comment by shengliu66** [comment]
I agree that the loss can reflect some information, but my point is that it is not "canceling" the CE loss, it reflects, maybe, how important the CE loss is compared with the regularization term. What you have shown is simply the loss of regularization term is larger, meaning maybe we are training more with the regularization and less with CE in this case. But the gradient from ELR is pointing to the direction for cleanly labeled examples, so this is fine. I'll close this issue as it is not very related to the code. We can discuss this through email.

**Comment by valencebond** [comment]
thanks for your explanation~
